src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -850,48 +862,53 @@ private static void CollectTargetTupleFields(NamedTypeSymbol underlying, ArrayBu
             }
         }
 
-        private SmallDictionary<Symbol, Symbol> UnderlyingDefinitionToMemberMap
+        internal SmallDictionary<Symbol, Symbol> UnderlyingDefinitionToMemberMap
         {
             get
             {
                 if (_lazyUnderlyingDefinitionToMemberMap == null)
                 {
-                    var map = new SmallDictionary<Symbol, Symbol>();
+                    var map = new SmallDictionary<Symbol, Symbol>(ReferenceEqualityComparer.Instance);
 
                     var underlyingDefinition = _underlyingType.OriginalDefinition;
                     var members = GetMembers();
-                    Symbol underlyingMemberDefinition;
 
-                    foreach (var member in members)
+                    // Go in reverse becaus we want members with default name, which precede the ones with
```

Comment:
Thanks. Will fix in the next change.

---
In reply to: [62271920](https://github.com/dotnet/roslyn/pull/11136#discussion_r62271920) [](ancestors = 62271920)